client/ReduxActions/account.js

```diff
@@ -1,66 +1,127 @@
+        APIParams: {
             url: '/api/account/login',
             type: 'POST',
             data: JSON.stringify({ username: auth.username, password: auth.password }),
+            contentType: 'application/json',
+            skipAuth: true
+        }
+    };
+}
+
+export function logoutAccount(tokenId) {
+    return {
+        types: ['LOGOUT_ACCOUNT', 'ACCOUNT_LOGGEDOUT'],
+        shouldCallAPI: () => true,
+        APIParams: {
+            url: '/api/account/logout',
+            type: 'POST',
+            data: JSON.stringify({ tokenId: tokenId }),
             contentType: 'application/json'
-        })
+        }
+    };
+}
+
+export function logout() {
+    return (dispatch, getState) => {
+        let state = getState();
+
+        if(!state.auth.refreshToken) {
+            return;
+        }
+
+        if(state.lobby.socket) {
+            state.lobby.socket.disconnect();
+        }
+
+        if(state.games.socket) {
+            state.game.socket.disconnect();
+        }
+
+        return dispatch(logoutAccount(state.auth.refreshToken.id));
     };
 }
 
 export function forgotPassword(details) {
     return {
         types: ['FORGOTPASSWORD_ACCOUNT', 'ACCOUNT_FORGOTPASSWORD'],
         shouldCallAPI: () => true,
-        callAPI: () => $.ajax({
+        APIParams: {
             url: '/api/account/password-reset',
             type: 'POST',
             data: JSON.stringify({ username: details.username, captcha: details.captcha }),
             contentType: 'application/json'
-        })
+        }
     };
 }
 
 export function resetPassword(details) {
     return {
         types: ['RESETPASSWORD_ACCOUNT', 'ACCOUNT_PASSWORDRESET'],
         shouldCallAPI: () => true,
-        callAPI: () => $.ajax({
+        APIParams: {
             url: '/api/account/password-reset-finish',
             type: 'POST',
             data: JSON.stringify({ id: details.id, token: details.token, newPassword: details.newPassword }),
             contentType: 'application/json'
-        })
+        }
     };
 }
 
 export function activateAccount(details) {
     return {
         types: ['ACTIVATE_ACCOUNT', 'ACCOUNT_ACTIVATED'],
         shouldCallAPI: () => true,
-        callAPI: () => $.ajax({
+        APIParams: {
             url: '/api/account/activate',
             type: 'POST',
             data: JSON.stringify({ id: details.id, token: details.token }),
             contentType: 'application/json'
-        })
+        }
+    };
+}
+
+export function setAuthTokens(token, refreshToken) {
+    return {
+        type: 'SET_AUTH_TOKENS',
+        token: token,
+        refreshToken: refreshToken
+    };
+}
+
+export function verifyAuthentication(token) {
+    return {
+        types: ['ACCOUNT_VERIFY_AUTH', 'ACCOUNT_AUTH_VERIFIED'],
+        shouldCallAPI: () => true,
+        APIParams: {
+            url: '/api/account/checkauth',
+            type: 'POST',
+            data: JSON.stringify({ token: token }),
```

Comment:
Is this necessary? If the user is authenticated (`state.auth.token` is set), then it should be automatically inserted into the headers. Likewise, `contentType` below should be handled automatically by the same middleware.

Reply:
Nope, wasn't even being used server side